tensorboard/webapp/app_routing/effects/app_routing_effects.ts

```diff
@@ -229,8 +229,27 @@ export class AppRoutingEffects {
         withLatestFrom(this.store.select(getActiveRoute)),
         filter(([, route]) => Boolean(route)),
         map(([navigatedAction, route]) => {
+          // DeepLinkProviders may tell AppRoutingEffects to modify URL query
+          // params upon redux state changes. These updates can discard URL
+          // hash storage, set by tf-storage on the Polymer side. Due to async
+          // operators, we cannot guarantee that the initial 'navigated' action
+          // will be processed before any other URL-modifying logic.
```

Comment:
I don't think it is purely about async operators. 

For instance, if you are navigated to timeseries (`#timeseries` is written by HashStorage) and if you interact with the pinning, this effect will overwrite and remove `#timeseries`. In this case, HashStorage was not even attempt to write any hash so there is nothing timing related about this.

Could you more accurately depict the problem please? 

Also I'd appreciate `hash storage` -> `HashStorageComponent`

Reply:
Updated the comment and dropped the part about async.